client/src/featureform/serving.py

```diff
@@ -143,112 +84,211 @@ def _host_features(self, features, entities):
+                feature_df = self.feature_df_with_entity(source['definition'], entity_id, feature)
+            feature_df_list.append(feature_df)
 
-            feature_row = self.sqldb.get_feature_variant(featureVariantTuple[0], featureVariantTuple[1])
-            entity_column, ts_column, feature_column_name, source_name, source_variant = feature_row['source_entity'], feature_row['source_timestamp'], feature_row['source_value'], feature_row['source_name'], feature_row['source_variant']
+        return feature_df_list
 
-            source_row = self.sqldb.get_source_variant(source_name, source_variant)
-            if self.sqldb.is_transformation(source_name, source_variant):
-                df = self.process_transformation(source_name, source_variant)
-                if isinstance(df, pd.Series):
-                    df = df.to_frame()
-                    df.reset_index(inplace=True)
-                df = df[[entity_tuple[0], feature_column_name]]
-                df.set_index(entity_tuple[0])
-                dataframe_mapping.append(df)
-            else:
-                name_variant = f"{featureVariantTuple[0]}.{featureVariantTuple[1]}"
-                dataframe_mapping = self.process_feature_csv(source_row['definition'], entity_tuple[0], entity_column,
-                                                             feature_column_name,
-                                                             dataframe_mapping,
-                                                             name_variant, ts_column)
+    def list_to_combined_df(self, features_list, entity_id):
+        all_feature_df = None
         try:
-            for value in dataframe_mapping:
+            for feature in features_list:
                 if all_feature_df is None:
-                    all_feature_df = value
+                    all_feature_df = feature
                 else:
-                    all_feature_df = all_feature_df.join(value.set_index(entity_tuple[0]), on=entity_tuple[0],
+                    all_feature_df = all_feature_df.join(feature.set_index(entity_id), on=entity_id,
                                                          lsuffix='_left')
+            return all_feature_df
         except TypeError:
             print("Set is empty")
-        entity_row = all_feature_df.loc[all_feature_df[entity_tuple[0]] == entity_tuple[1]].copy()
-        entity_row.drop(columns=entity_tuple[0], inplace=True)
-        if len(entity_row.values) > 0:
-            return entity_row.values[0]
+
+    def get_features_for_entity(self, entity_id, entity_value, all_feature_df):
+        entity = all_feature_df.loc[all_feature_df[entity_id] == entity_value].copy()
+        entity.drop(columns=entity_id, inplace=True)
+        if len(entity.values) > 0:
+            return entity.values[0]
         else:
-            raise Exception("No matching entities for {}".format(entity_tuple))
+            raise Exception(f"No matching entities for {entity_id}: {entity_id}")
```

Comment:
Is the second value suppose to be something else?

Reply:
Yup, changed it